skbeam/io/tests/test_fit2d_save.py

```diff
@@ -44,17 +44,24 @@
 import numpy as np
 from skbeam.io.fit2d import fit2d_save, read_fit2d_msk
 from numpy.testing import assert_array_equal
-import pytest
+import sys
+
+if sys.version_info >= (3, 0):
+    from tempfile import TemporaryDirectory
+else:
+    from backports.tempfile import TemporaryDirectory
+
 
-@pytest.mark.xfail(raises=IOError)
 def test_save_output_fit2d():
```

Comment:
There's a built-in fixture in pytest that handles temporary directory creation. pass `tmpdir` as a function argument and pytest will provide that fixture for you. Use it like so:

You can create a file with tmpdir by calling `tmpdir.join('file name')` which will produce an object with some things hanging off of it. The most useful ones in this context are `.dirname` and `.filename`
```python
t = tmpdir.join('file_name')
dirname = t.dirname
filename = t.strpath
```

or in full:

```python
from __future__ import absolute_import, division, print_function
import os
import numpy as np
from skbeam.io.fit2d import fit2d_save, read_fit2d_msk
from numpy.testing import assert_array_equal
import sys


def test_save_output_fit2d(tmpdir):
    t = tmpdir.join('function_values')
    dir_path = t.dirname
    filename = t.strpath
    msk = np.random.random_integers(
        0, 1, (np.random.random_integers(0, 200),
               np.random.random_integers(0, 200))).astype(bool)

    fit2d_save(msk, filename, dir_path=dir_path)
    msk2 = read_fit2d_msk('%s.msk' % filename)
    assert_array_equal(msk2, msk)
```

Then you don't need the backports or py2/py3 compatibility